<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
  
  <!-- This xacro provides a ros2_control configuration for FakeRobotHardware -->
  <xacro:macro name="fake_hardware_ros2_control" params="name">
    <ros2_control name="${name}" type="system">
      <hardware>
        <plugin>pick_place_hardware/FakeRobotHardware</plugin>
        <param name="noise_level">0.001</param>
        <param name="actuator_delay">0.05</param>
        <param name="position_gain">5.0</param>
      </hardware>

      <joint name="j1">
        <command_interface name="position"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>

      <joint name="j2">
        <command_interface name="position"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>

      <joint name="j3">
        <command_interface name="position"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>

      <joint name="j4">
        <command_interface name="position"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>

      <joint name="j5">
        <command_interface name="position"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>

      <joint name="j6">
        <command_interface name="position"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>

      <joint name="j7l">
        <command_interface name="position"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>

      <joint name="j7r">
        <command_interface name="position"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>

    </ros2_control>
  </xacro:macro>

</robot>
